File: backend/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  product_name: {
    type: String,
    required: true,
    trim: true
  },
  brand_name: {
    type: String,
    required: true,
    trim: true
  },
  brand_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Brand',
    required: true
  },
  product_image: {
    type: String,
    required: true // Path to the image file
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Product', productSchema);

File: android/app/src/main/java/com/shoecatalog/app/data/model/Product.kt
package com.shoecatalog.app.data.model

import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("_id") val id: String,
    @SerializedName("product_name") val productName: String,
    @SerializedName("brand_name") val brandName: String,
    @SerializedName("product_image") val productImage: String
)

